nemo_automodel/checkpoint/checkpointing.py

```
@@ -77,12 +95,14 @@ def save_model(
         weights_path: Path to save model weights
         checkpoint_config: Checkpointing configuration
     """
-    # TODO(@adil-a): Need to add support for PEFT.
+    if checkpoint_config.is_peft and checkpoint_config.model_save_format == SerializationFormat.TORCH_SAVE:
```

I would throw an exception when the checkpoint_config is created instead of logging here.